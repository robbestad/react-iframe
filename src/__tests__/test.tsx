import React from "react"
import Iframe from "../iframe"
import renderer from "react-test-renderer"

test("Render default iframe", () => {
	const component = renderer.create(
		<Iframe position="relative" url="http://www.foo.com"  src="http://www.foobar.com"/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

test("Render iframe with title", () => {
	const component = renderer.create(
		<Iframe position="relative"
						title="A foobared iframe"
						url="http://www.foobar.com"
						src="http://www.foo.com"
						ariaHidden={true}
						ariaLabel="someLabel"
						ariaLabelledby="someId"
						allow="encrypted-media"
						sandbox="allow-scripts"
						allowFullScreen={false}
						className="myIframeClass"
						frameBorder={0}
						overflow="auto"
						loading="eager"
						onLoad={()=>{console.log("hello")}}
						onMouseOut={()=>{console.log("goodbye")}}
						onMouseOver={()=>{console.log("hi")}}
						display="block"
						id="myIframeId"
						name="My Iframe name"
						scrolling="yes"
						styles={{background: "red"}}
						target="_self"
		/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})


test("Render with allowFullScreen bool", () => {
	const component = renderer.create(
		<Iframe allowFullScreen={false}
				allow="fullscreen"
				frameBorder={1} url="http://www.foobar.com"/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})


test("Render with allow attribute", () => {
	const component = renderer.create(
		<Iframe allowFullScreen={true}
				allow="fullscreen autoplay"
				frameBorder={1} url="http://www.foobar.com"/>
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})
