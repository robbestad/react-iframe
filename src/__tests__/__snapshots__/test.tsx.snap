// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default iframe 1`] = `
<iframe
  allowFullScreen="allowFullScreen"
  src="http://www.foobar.com"
  style={
    Object {
      "display": "initial",
      "position": "relative",
    }
  }
/>
`;

exports[`Render iframe with title 1`] = `
<iframe
  allow="encrypted-media"
  allowFullScreen="allowFullScreen"
  aria-hidden={true}
  aria-label="someLabel"
  aria-labelledby="someId"
  className="myIframeClass"
  id="myIframeId"
  loading="eager"
  name="My Iframe name"
  onLoad={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  sandbox="a l l o w - s c r i p t s"
  scrolling="yes"
  src="http://www.foo.com"
  style={
    Object {
      "background": "red",
      "border": 0,
      "display": "block",
      "overflow": "auto",
      "position": "relative",
    }
  }
  target="_self"
  title="A foobared iframe"
/>
`;

exports[`Render with allow attribute 1`] = `
<iframe
  allow="fullscreen autoplay"
  allowFullScreen="allowFullScreen"
  src="http://www.foobar.com"
  style={
    Object {
      "border": 1,
      "display": "initial",
    }
  }
/>
`;

exports[`Render with allowFullScreen bool 1`] = `
<iframe
  allow="fullscreen"
  allowFullScreen="allowFullScreen"
  src="http://www.foobar.com"
  style={
    Object {
      "border": 1,
      "display": "initial",
    }
  }
/>
`;
